src/scenes/Player.ts

```diff
@@ -2,5 +2,16 @@ import { Graph } from "graphlib";
 import { IPlayer } from "./IPlayer";
 
 export class Player implements IPlayer {
-    constructor(private graph: Graph, private location: string) {}
+    constructor(private graph: Graph, private location: string) {
+        this.graph = graph;
+        this.location = location;
+    }
+
+    public setLocation(city: string) {
```

Comment:
consider using an `public setLocation(city: CityName) {` over string variables.

Reply:
The `enum` provides you with a lot more type-safety thus improving development speed

Reply:
multiple occurrences of this